Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java

```diff
@@ -2,44 +2,36 @@
 
 import android.content.Context;
 import android.content.Intent;
-import android.graphics.Color;
+import android.content.SharedPreferences;
 import android.os.Bundle;
+import android.preference.PreferenceManager;
 import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
-import android.text.format.DateFormat;
-import android.util.Log;
 import android.view.MenuItem;
-import android.widget.ImageView;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Spinner;
 import android.widget.TextClock;
-import android.widget.TextView;
-
+import java.util.Arrays;
 import java.util.Calendar;
+import java.util.Date;
 import java.util.Objects;
 import java.util.TimeZone;
-
-import butterknife.BindView;
 import butterknife.ButterKnife;
-import butterknife.OnClick;
 import io.github.project_travel_mate.R;
-import utils.CurrencyConverterGlobal;
 
 public class WorldClockActivity extends AppCompatActivity {
 
-    @BindView(R.id.timezone_country_flag)
-    ImageView from_image;
-    @BindView(R.id.timezone_country_name)
-    TextView timezone_name;
-    Boolean flag_check_first_item = false;
-    private Context mContext;
-    String timezone_short = "Asia/Kolkata";
-    private AnalogClock mAnalogClock;
-    private String mFormat;
-    private final String mHours12 = "hh:mm:ss a";
-    private final String mHours24 = "k:mm:ss";
-
-    //private DigitalClock mDigitalClock;
-
-    private static TextClock mDigitalClock;
+    private CustomAnalogClock mAnalogClock;
+    private static final String DEFAULT_TIME_ZONE_KEY = "defaultTimeZone";
+    private TextClock mTextClock;
+    private long mMiliSeconds;
+    private ArrayAdapter<String> mIdAdapter;
+    private Spinner mSpinnerAvailableID;
```

Comment:
what does this spinner indicate? It's not clear from name

Reply:
This spinner shows the available time zones for the users.

Reply:
not clear from var name. make it verbose